File: src/main/java/com/beijunyi/hppc/api/AdminApi.java
package com.beijunyi.hppc.api;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;
import javax.inject.Named;
import javax.inject.Singleton;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.core.Context;
import javax.ws.rs.core.Response;
import javax.ws.rs.core.UriInfo;

import com.beijunyi.hppc.models.api.*;

@Named
@Singleton
@Path("/api/admin")
public class AdminApi {

  @GET
  @Path("accounts")
  public Response queryAccounts(@Nonnull @Context UriInfo uriInfo) {
    return Response.ok().build();
  }

  @POST
  @Path("account/create")
  public Response createAccount(@Nullable UpdateAccountRequest request) {
    return Response.status(Response.Status.BAD_REQUEST).build();
  }

  @POST
  @Path("account/update")
  public Response updateAccount(@Nullable UpdateAccountRequest request) {
    return Response.status(Response.Status.BAD_REQUEST).build();
  }

  @POST
  @Path("account/change-password")
  public Response changeAccountPassword(@Nullable ChangeAccountPasswordRequest request) {
    return Response.status(Response.Status.BAD_REQUEST).build();
  }

  @POST
  @Path("account/change-role-mapping")
  public Response changeAccountRoleMapping(@Nullable ChangeAccountRoleMappingRequest request) {
    return Response.status(Response.Status.BAD_REQUEST).build();
  }

  @GET
  @Path("roles")
  public Response queryRoles(@Nonnull @Context UriInfo uriInfo) {
    return Response.ok().build();
  }

  @POST
  @Path("role/create")
  public Response createRole(@Nullable UpdateRoleRequest request) {
    return Response.status(Response.Status.BAD_REQUEST).build();
  }

  @POST
  @Path("role/update")
  public Response updateRole(@Nullable UpdateRoleRequest request) {
    return Response.status(Response.Status.BAD_REQUEST).build();
  }

  @POST
  @Path("role/change-privilege-mapping")
  public Response changeRolePrivilegeMapping(@Nullable ChangeRolePrivilegeMappingRequest request) {
    return Response.status(Response.Status.BAD_REQUEST).build();
  }

  @GET
  @Path("privileges")
  public Response queryPrivileges(@Nonnull @Context UriInfo uriInfo) {
    return Response.ok().build();
  }

  @POST
  @Path("privilege/create")
  public Response createPrivilege(@Nullable UpdatePrivilegeRequest request) {
    return Response.status(Response.Status.BAD_REQUEST).build();
  }

  @POST
  @Path("privilege/update")
  public Response updatePrivilege(@Nullable UpdatePrivilegeRequest request) {
    return Response.status(Response.Status.BAD_REQUEST).build();
  }

}

File: src/main/webapp/scripts/modules/Admin/AdminApi.js
app.service('AdminApi', function($http) {
  return {
    queryAccounts: function(params) {
      return $http.get('/api/admin/accounts', {params: params});
    },
    createAccount: function(account) {
      return $http.post('/api/admin/account/create', account);
    },
    updateAccount: function(account) {
      return $http.post('/api/admin/account/update', account);
    },
    changeAccountPassword: function(request) {
      return $http.post('/api/admin/account/change-password', request);
    },
    changeAccountRoleMapping: function(request) {
      return $http.post('/api/admin/account/change-role-mapping', request);
    },
    queryRoles: function(params) {
      return $http.get('/api/admin/roles', {params: params});
    },
    createRole: function(role) {
      return $http.post('/api/admin/role/create', role);
    },
    updateRole: function(role) {
      return $http.post('/api/admin/role/update', role);
    },
    changeRolePrivilegeMapping: function(request) {
      return $http.post('/api/admin/role/change-privilege-mapping', request);
    },
    queryPrivileges: function(params) {
      return $http.get('/api/admin/privileges', {params: params});
    },
    createPrivilege: function(privilege) {
      return $http.post('/api/admin/privilege/create', privilege);
    },
    updatePrivilege: function(privilege) {
      return $http.post('/api/admin/privilege/update', privilege);
    }
  };
});